### ISSUE TITLE
PCILeech_VmmRead DLL - cb and pb variables

### ISSUE THREAD

**Issue by Flodkozza** [ufrisk/pcileech#59]
I see that cb is a DWORD and pb is PBYTE. I think that pb is where to store the value and cb is how many bytes to read? If I'm correct, how can I read an int or float since it has to be PBYTE? I tried to convert the bytes to int but I can't get it to work.

**Comment by Flodkozza** [comment]
Thanks, by the way do you know if there is any online shop that ships to any country in Europe that sells SP605? Or I should just get the Pciescreamer? If it helps, I have about 20 reads in an infinite loop and they are all 4 bytes each. And does the Pciescreamer already come with the soldered on switch for external power/pcie power and the jtag connector? Since in their shop, the pictures are showing it without them so I'm not sure where to get the things that need to be soldered.

**Comment by ufrisk** [comment]
digikey.com ships to europe even if american, but you'll have to pay VAT on the imports. DHL always bill me for it. Just make sure you get the USB3 addon board as well that I have listed on my page.

The JTAG connector is soldered on the PCIeScreamer board when purchased.

But if just trying out the VMM it's possible to do it in read-only mode with memory dump files - no need for hardware. Or in read-write mode if you do it on Windows 7 only patched with the January patches (== vulnerable for TotalMeltdown - CVE-2018-1038).

**Comment by Flodkozza** [comment]
Oh yeah I totally forgot the memory dump files. I will make sure I try it before I purchase. Tho I live outside of European Union but in Europe, so there is no VAT ( There are another import duties and taxes tho of course.. ) I have like 550€ in cash ready for the board so I guess I have to go for the PCIeScreamer since I'll have to pay the import duties as well that are gonna be like 150€. Sucks lol.

**Comment by Flodkozza** [comment]
```
CALL:    PCILeech_InitializeFromFile
SUCCESS: PCILeech_InitializeFromFile
CALL:    PCILeech_VmmInitialize
FAIL:    PCILeech_VmmInitialize
```
Initializing the Vmm fails tho.

**Comment by ufrisk** [comment]
1 make sure you are targeting windows 64-bit version
2 make sure it's a dump file of raw memory dump format (flat dump file, not microsoft minidump file or similar).
3 run `pcileech.exe identify -file <yourfile>` to try to identify the -cr3 value optionally. you may use this one in the pcileech calls to identify kernel page table base (sometimes it fails to auto-detect it).

**Comment by Flodkozza** [comment]
Target is Windows 10 64-bit.
My dump file is raw memory dump.
My Windows is not activated since it's a pirated version and I'm lazy to activate it so could that be a problem?

**Comment by ufrisk** [comment]
no that's not it, pirated versions should have absolutely zero affect on this. try run pcileech.exe identify as above to see if it works.

also if running hyper-v or credential guard or anything of that memory dumping sometimes fail with corrupt on newer windows 10's. also, I have not verified this on the most recent windows insider (beta) builds if you are running that.

**Comment by pekiko** [comment]
Hi, I see you are using pcileech dll. Did you also need to download pcileech.exe and all the headers and files or only dll and dll.h? Thanks

**Comment by Flodkozza** [comment]
Okay, so I just went out yesterday and ordered the Pciescreamer, expected delivery is tomorrow so I will update here if everything goes smoothly. Btw if I want to change the device id, that is done in the Vivado project settings where the vendor id etc are changed? Like it's stated in the Advanced Building readme? I tried changing it but the project wouldn't build, but if I tried building the original files it builds fine.
